pandas/core/generic.py

```diff
@@ -6005,20 +6005,25 @@ def fillna(
                 )
 
             elif isinstance(value, (dict, ABCSeries)):
+                new_data = self.copy()
```

Comment:
copy / transpose are both potential very expensive operations - is there a way to do this without either of those?

Reply:
@WillAyd 
Okay. copy conditional. remove transpose.